File: backend/src/main/java/com/java/NBE4_5_1_7/domain/study/repository/StudyMemoLikeRepository.java
package com.java.NBE4_5_1_7.domain.study.repository;

import com.java.NBE4_5_1_7.domain.study.entity.StudyMemo;
import com.java.NBE4_5_1_7.domain.study.entity.StudyMemoLike;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface StudyMemoLikeRepository extends JpaRepository<StudyMemoLike, Integer> {
    int countByStudyMemoId(Long studyMemoId);

    Optional<StudyMemoLike> findByStudyMemo(StudyMemo studyMemo);
}

File: backend/src/main/java/com/java/NBE4_5_3_7/domain/study/repository/StudyMemoLikeRepository.kt
package com.java.NBE4_5_3_7.domain.study.repository

import com.java.NBE4_5_3_7.domain.member.entity.Member
import com.java.NBE4_5_3_7.domain.study.entity.StudyMemo
import com.java.NBE4_5_3_7.domain.study.entity.StudyMemoLike
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface StudyMemoLikeRepository : JpaRepository<StudyMemoLike?, Int?> {
    fun countByStudyMemoId(studyMemoId: Long?): Int

    fun findByStudyMemoAndMember(studyMemo: StudyMemo?, member: Member): Optional<StudyMemoLike?>?
}
